test/kubernetes/kubernetes.go

```diff
@@ -119,3 +135,121 @@ func CleanupNamespaces(c kubernetes.Interface) error {
 
 	return nil
 }
+
+// NewIngress creates a new ingress
+func NewIngress(c kubernetes.Interface, ingress *networking.Ingress) error {
+	if ingress.Namespace != metav1.NamespaceNone && ingress.Namespace != metav1.NamespaceDefault {
+		return fmt.Errorf("Ingress definitions in the default namespace are not allowed")
+	}
+
+	_, err := c.NetworkingV1().Ingresses(ingress.Namespace).Create(context.TODO(), ingress, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("creating Ingress: %w", err)
+	}
+
+	return nil
+}
+
+// IngressFromSpec deserializes an Ingress definition using an IngressSpec
+func IngressFromSpec(name, namespace, ingressSpec string) (*networking.Ingress, error) {
+	if namespace != metav1.NamespaceNone && namespace != metav1.NamespaceDefault {
+		return nil, fmt.Errorf("Ingress definitions in the default namespace are not allowed")
+	}
+
+	ingress := &networking.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	if err := yaml.Unmarshal([]byte(ingressSpec), &ingress.Spec); err != nil {
+		return nil, fmt.Errorf("deserializing Ingress from spec: %w", err)
+	}
+
+	return ingress, nil
+}
+
+// IngressFromManifest deserializes an Ingress definition using an Ingress
+func IngressFromManifest(namespace, manifest string) (*networking.Ingress, error) {
+	if namespace != metav1.NamespaceNone && namespace != metav1.NamespaceDefault {
+		return nil, fmt.Errorf("Ingress definitions in the default namespace are not allowed")
+	}
+
+	ingress := &networking.Ingress{}
+	if err := yaml.Unmarshal([]byte(manifest), &ingress); err != nil {
+		return nil, fmt.Errorf("deserializing Ingress from manifest: %w", err)
+	}
+
+	ingress.SetNamespace(namespace)
+	return ingress, nil
+}
+
+const (
+	// waitForIngressAddressTimeout wait time for valid ingress status
+	waitForIngressAddressTimeout = 5 * time.Minute
```

Comment:
we should probably make this parametrizable given different implementations (e.g. you may know that your impl works quickly

Reply:
done